lib/templates/accountTemplate.js

```diff
@@ -107,21 +107,69 @@ class AccountTemplate {
     }, {});
 
     // We need to also create the mapping_list_ranges, which could be different per firm & partner
+
+    /* 
+      Mapping list ranges sample in config.json:
+      ...
+    "account_range": "280,282", // This is the "no mapping list" range
+    "mapping_list_ranges": [
+      {
+        "partner_account_mapping_list_id": 2,
+        "account_range": "280,282",
+        "type": "partner",
+        "env_id": "84"
+      },
+      {
+        "account_mapping_list_id": 8957,
+        "account_range": "280,282,284",
+        "type": "firm",
+        "env_id": "13827"
+      }
+    ],
+    ...
+    */
+
     const existingMappingRanges = existingConfig.mapping_list_ranges || [];
-    const updatedMappingRanges = [...existingMappingRanges];
+    const mappingListIds = template.mapping_list_ranges.map((range) =>
+      type == "partner"
+        ? range.partner_account_mapping_list_id
+        : range.account_mapping_list_id
+    );
+
+    let updatedMappingRanges = [...existingMappingRanges];
+
+    // Remove the mapping list ranges from this partner / firm env_id which are no longer part of the response
+    updatedMappingRanges = updatedMappingRanges.filter(
+      (range) =>
+        mappingListIds.includes(
+          type == "partner"
+            ? range.partner_account_mapping_list_id
+            : range.account_mapping_list_id
+        ) || range.env_id !== envId
```

Comment:
I suppose env_id is either firm_id or partner_id. Based on this last check, it means all mapping list ranges, for all partners and firms, are stored on the same config file, right?

Reply:
I'm just worried that file might eventually get too big that it takes a long time to open. But probably can be refactored later